doc/widgets.lean

```diff
@@ -0,0 +1,203 @@
+import Lean
+open Lean Widget
+
+/-!
+# The user-widgets system
+
+Proving and programming are inherently interactive tasks. Lots of mathematical objects and data
+structures are visual in nature. *User widgets* let you associate custom interactive UIs with
+sections of a Lean document. User widgets are rendered in the Lean infoview.
+
+![Rubik's cube](images/widgets_rubiks.png)
```

Comment:
The image seems to be still showing the old code with `@[staticJS]`

Reply:
I think that's okay, it is there simply to give an idea of "type stuff here, UI appears there" rather than serve as an API reference.